### PR TITLE
refactor: move get settings [part 3]

### PR DESCRIPTION
**Pull request by glevco**
Depends on https://github.com/HathorNetwork/hathor-core/pull/768

### Motivation

Please read the description on https://github.com/HathorNetwork/hathor-core/pull/768, this PR is just a follow-up part.

### Acceptance Criteria

- Remove some usages of `HathorSettings()` from the module level and move it to inside functions/methods
  - For functions, the function just calls the new `get_settings()`
  - For classes we set `self._settings = get_settings()` in the class' `__init__()`, and update all methods to use that new attribute

### Checklist

- [x] If you are requesting a merge into `master`, confirm this code is production-ready and can be included in future releases as soon as it gets merged

### CODE REVIEW

File: hathor/p2p/states/ready.py [hunk 1]
```diff
@@ -209,10 +208,11 @@ def handle_pong(self, payload: str) -> None:
         self.ping_start_time = None
         self.log.debug('rtt updated', rtt=self.ping_rtt, min_rtt=self.ping_min_rtt)
 
-    def send_get_best_blockchain(self, n_blocks: int = settings.DEFAULT_BEST_BLOCKCHAIN_BLOCKS) -> None:
+    def send_get_best_blockchain(self, n_blocks: Optional[int] = None) -> None:
         """ Send a GET-BEST-BLOCKCHAIN command, requesting a list of the latest
         N blocks from the best blockchain.
         """
+        n_blocks = n_blocks if n_blocks is not None else self._settings.DEFAULT_BEST_BLOCKCHAIN_BLOCKS
```
Comment (msbrogli) [inline hathor/p2p/states/ready.py:215]: Use explicit typing here: `n_blocks: int`.

File: hathor/wallet/util.py [hunk 2]
```diff
@@ -72,6 +71,8 @@ def generate_multisig_address(redeem_script: bytes, version_byte: bytes = settin
         :return: The multisig address
         :rtype: str(base58)
     """
+    settings = get_settings()
+    version_byte = version_byte if version_byte is not None else settings.MULTISIG_VERSION_BYTE
```
Comment (msbrogli) [inline hathor/wallet/util.py:75]: Same here.

**Review by msbrogli — APPROVED** [review]

**Review by jansegre — DISMISSED** [review]

File: hathor/p2p/states/ready.py (hunk 1 above)
Comment (glevco) [reply hathor/p2p/states/ready.py]: Changed in https://github.com/HathorNetwork/hathor-core/pull/770/commits/e2afcd970cff9597b7d5c5528b09e3a971870fc6. I had to change the variable name for the linter

File: hathor/wallet/util.py (hunk 2 above)
Comment (glevco) [reply hathor/wallet/util.py]: Changed in https://github.com/HathorNetwork/hathor-core/pull/770/commits/e2afcd970cff9597b7d5c5528b09e3a971870fc6. I had to change the variable name for the linter

**Review by msbrogli — APPROVED** [review]

**Review by jansegre — APPROVED** [review]

File: hathor/p2p/states/ready.py (hunk 1 above)
Comment (msbrogli) [reply hathor/p2p/states/ready.py]: Thanks!

File: hathor/wallet/util.py (hunk 2 above)
Comment (msbrogli) [reply hathor/wallet/util.py]: Thanks!

**Pull request merged**